### PR TITLE
fix: remove editor text persistence

### PR DESCRIPTION
**Pull request by awais-ansari**
### [INF-589](https://2u-internal.atlassian.net/browse/INF-589)

- Text Editor will reset to the initial value then click on submit and cancel button while creating the post.
- Text Editor will reset to the initial value then click on submit and cancel button while creating the Response/Comment.
- Text Editor will not persist the text.

https://www.loom.com/share/896bd04092c1422aa864a8cd64e6256e

### CODE REVIEW

**Review by muhammadadeeltajamul — APPROVED** [review]

File: src/index.scss [hunk 1]
```diff
@@ -227,7 +227,7 @@ header {
 
 .header-action-bar {
   background-color: #fff;
-  z-index: 1;
+  z-index: 2;
```
Comment (saadyousafarbi) [inline src/index.scss:230]: why is this changed?

File: src/discussions/posts/post-editor/PostEditor.jsx [hunk 2]
```diff
@@ -188,18 +201,6 @@ function PostEditor({
...
-    topic: post?.topicId || topicId || nonCoursewareTopics?.[0]?.id,
-    title: post?.title || '',
-    comment: post?.rawBody || '',
-    follow: isEmpty(post?.following) ? true : post?.following,
-    anonymous: allowAnonymous ? false : undefined,
-    anonymousToPeers: allowAnonymousToPeers ? false : undefined,
-    editReasonCode: post?.lastEdit?.reasonCode || '',
-    cohort: post?.cohort || 'default',
-  };
-
```
Comment (mehaknasir) [inline src/discussions/posts/post-editor/PostEditor.jsx:202]: what happens? when you open the editor to edit posts/comments/responses?

File: src/components/TinyMCEEditor.jsx [hunk 3]
```diff
@@ -92,7 +92,7 @@ export default function TinyMCEEditor(props) {
         browser_spellcheck: true,
         a11y_advanced_options: true,
         autosave_interval: '1s',
-        autosave_restore_when_empty: true,
+        autosave_restore_when_empty: false,
```
Comment (mehaknasir) [inline src/components/TinyMCEEditor.jsx:95]: why mark it as false when you can reset the form on cancel/submit?

**Review by mehaknasir — COMMENTED** [review]
I feel like these changes will create more issues, like what if someone wrote the post of 100 lines and cancel it by mistake?  and similar usecases.

File: src/discussions/posts/post-editor/PostEditor.jsx (hunk 2 above)
Comment (awais-ansari) [reply src/discussions/posts/post-editor/PostEditor.jsx:202]: It will work as it was working before. I just relocate this constant.

File: src/components/TinyMCEEditor.jsx (hunk 3 above)
Comment (awais-ansari) [reply src/components/TinyMCEEditor.jsx:95]: Yes, I'm resetting the form state on cancel/submit but This plugin stores the draft value in browser storage. If it is true it will get the draft value from the browser storage after the editor initialization.

File: src/index.scss (hunk 1 above)
Comment (awais-ansari) [reply src/index.scss:230]: The text editor action bar and header action bar had the same z-index value `1` which can create unpredictable UI overlap glitches for some browsers that why I increase this value according to UI stack.

**Review by mehaknasir — APPROVED** [review]
